Added note about moz-extension://  URLs in tabs.query()

snoack opened the pull request:
For reference: https://discourse.mozilla.org/t/moz-extension-not-allowed-in-tabs-query/12793

snoack commented:
This is an inconsistency in Firefox, other browsers don't error out if `url` isn't a (valid) match pattern. But you are right that since the documentation already specifies (incorrectly) that a match pattern is expected, this is redundant. So I suggest removing this part from the documentation.

wbamberg commented:
I'd like the "correct" version to be in the docs, and deviations noted in the compat data. Which version is "correct" is not that obvious here, but [Chrome's docs](https://developer.chrome.com/extensions/tabs#method-query), [the standard](https://browserext.github.io/browserext/#dom-browserexttabquerydetails-url), and the MDN docs all say that `url` is a match pattern.

So I think it would be best to keep the docs as they are and add a note for Chrome, Opera (and Edge?) to say that other values are also allowed in those browsers.

snoack commented:
Sorry, I just tested on Chrome again, and it turned out that in fact, it needs to be a valid match pattern there as well. So the only incompatibility is that before Firefox 56, `moz-extension://` URLs were not allowed here.

Also I noticed that the `url` property is ignored on Chrome as well, if the extension doesn't have the  `tabs` permission (see the [docs](https://developer.chrome.com/extensions/tabs#property-queryInfo-url)). So documenting this as an exception for Firefox seems incorrect.

I updated the patch accordingly.

wbamberg commented:
> Sorry, I just tested on Chrome again, and it turned out that, in fact, a match pattern is expected there as well. So the only incompatibility is that before Firefox 56, `moz-extension://` URLs were not allowed here.

Thanks for testing!

> Also I noticed that the `url` property is ignored on Chrome as well, if the extension doesn't have the  `tabs` permission (see the [docs](https://developer.chrome.com/extensions/tabs#property-queryInfo-url)). So documenting this as an exception for Firefox seems incorrect.

That's interesting, and new: https://bugzilla.mozilla.org/show_bug.cgi?id=1237822.

snoack commented:
> I would prefer to omit this note, and document this only in the match-pattern docs. AIUI there are other places where this matters (e.g. webRequest), and I'd prefer to document it in one place.

I'm afraid that nobody will find it in the match pattern docs. But you are right, that request filtering is also effected, though this is a rather theoretical scenario, since the extension isn't supposed to see requests by other extensions anyway, and there is not much of a point to intercept its own requests. Anyway, I added a note there as well.

> That's interesting, and new: https://bugzilla.mozilla.org/show_bug.cgi?id=1237822.

I didn't actually tested it before, but naively assumed that the Chrome documentation is accurate. Anyway, I just tested it myself, on Chrome 61, and in fact `tabs.query()` completes successful, without `tabs` permission, while Firefox fails with an asynchronous error. So I added back a note about Firefox here.
